pkg/controller/bastion/actuator_reconcile.go

```diff
@@ -0,0 +1,291 @@
+			Tags:               &compute.Tags{Items: []string{opt.BastionInstanceName}},
+			Metadata:           &compute.Metadata{Items: metadataItems},
+		}
+
+		_, err = gcpclient.Instances().Insert(opt.ProjectID, opt.Zone, instance).Context(ctx).Do()
+		if err != nil {
+			return nil, fmt.Errorf("%w, failed to create instance", err)
+		}
+	}
+
+	return getInstanceEndpoints(ctx, gcpclient, opt)
+}
+
+func getInstanceEndpoints(ctx context.Context, gcpclient gcpclient.Interface, opt *Options) (*bastionEndpoints, error) {
+	instance, err := getBastionInstance(ctx, gcpclient, opt)
+	if err != nil {
+		return nil, err
+	}
+
+	if instance == nil {
+		return nil, nil
+	}
+
+	if instance.Status != "RUNNING" {
+		return nil, fmt.Errorf("Instance not RUNNING, Status:" + instance.Status)
+
+	}
+
+	endpoints := &bastionEndpoints{}
+
+	networkInterfaces := instance.NetworkInterfaces
+	internalIP := &networkInterfaces[0].NetworkIP
+	externalIP := &networkInterfaces[0].AccessConfigs[0].NatIP
+
+	if len(networkInterfaces) == 0 {
+		return nil, fmt.Errorf(instance.Name + ":" + "no network interfaces found")
+	}
+
+	if len(networkInterfaces[0].AccessConfigs) == 0 {
+		return nil, fmt.Errorf(instance.Name + ":" + "no access config found for network interface")
+	}
+
+	if ingress := addressToIngress(&instance.Name, internalIP); ingress != nil {
+		endpoints.private = ingress
+	}
+
+	// public dns name not supported now
+	// https://cloud.google.com/compute/docs/instances/create-ptr-record#api
+	if ingress := addressToIngress(nil, externalIP); ingress != nil {
+		endpoints.public = ingress
+	}
+
+	return endpoints, nil
+}
+
+// bastionEndpoints collects the endpoints the bastion host provides; the
+// private endpoint is important for opening a port on the worker node
+// ingress firewall rule to allow SSH from that node, the public endpoint is where
+// the enduser connects to to establish the SSH connection.
+type bastionEndpoints struct {
+	private *corev1.LoadBalancerIngress
+	public  *corev1.LoadBalancerIngress
+}
+
+// Ready returns true if both public and private interfaces each have either
+// an IP or a hostname or both.
+func (be *bastionEndpoints) Ready() bool {
+	return be != nil && IngressReady(be.private) && IngressReady(be.public)
+}
+
+// IngressReady returns true if either an IP or a hostname or both are set.
+func IngressReady(ingress *corev1.LoadBalancerIngress) bool {
+	return ingress != nil && (ingress.Hostname != "" || ingress.IP != "")
+}
+
+// addressToIngress converts the IP address into a
+// corev1.LoadBalancerIngress resource. If both arguments are nil, then
+// nil is returned.
+func addressToIngress(dnsName *string, ipAddress *string) *corev1.LoadBalancerIngress {
+	var ingress *corev1.LoadBalancerIngress
+
+	if ipAddress != nil || dnsName != nil {
+		ingress = &corev1.LoadBalancerIngress{}
+		// public dns name not supported now
+		// https://cloud.google.com/compute/docs/instances/create-ptr-record#api
```

Comment:
```suggestion
```
comment does not fit in this util function

Reply:
committed in fbad5cfe22865b73a53ff0684b28bd8563af8e5d